src/main/lrsql/ops/command/admin.clj

```diff
@@ -44,3 +45,32 @@
       (bp/-delete-admin-account! bk tx input)
       {:result (:account-id input)})
     {:result :lrsql.admin/missing-account-error}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Ensure Admin Account from OIDC
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/fdef ensure-admin-oidc!
+  :args (s/cat :bk ads/admin-backend?
+               :tx transaction?
+               :input ads/ensure-admin-oidc-input-spec)
+  :ret ads/ensure-admin-ret-spec)
+
+(defn ensure-admin-oidc!
+  "Create a new admin with OIDC issuer or verify issuer of an existing admin.
+  Returns a map where `:result` is the account ID."
+  [bk tx {:keys [username oidc-issuer]
+          :as   ensure-input}]
+  (if-let [{extant-issuer :oidc_issuer ;; TODO: is not coerced to kebab?
+            id            :id} (bp/-query-account-oidc
+                                bk tx {:username username})]
+    {:result
+     (if (= oidc-issuer extant-issuer)
+       id
+       :lrsql.admin/oidc-issuer-mismatch-error)}
+    (let [{:keys [primary-key]
+           :as   insert-input} (admin-i/insert-admin-oidc-input
+                                ensure-input)]
+      (do
```

Comment:
This `do` is unneeded.

Reply:
https://github.com/yetanalytics/lrsql/pull/207/commits/0d2cfe97791fa990b600eaef06e9ac19ab7dfcab